### PR TITLE
Add basic Rust bindings

### PR DESCRIPTION
**Pull request by axic**
Part of #168.

### CODE REVIEW

**Comment by axic** [comment]
Next step is to create a `Result` type which wraps a `u8` array.

This type should:
- a `from` trait from a `u8[]` slice
- a function which returns a C function for destroying the instance (the C signature is `extern fn evmc_release_result(result: *const evmc_result)` where `result` is a pointer to object)

File: bindings/rust/evmc-sys/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "evmc-sys"
+version = "6.2.0-dev"
+authors = ["Alex Beregszaszi <alex@rtfs.hu>"]
+license = "Apache-2.0"
+repository = "https://github.com/ethereum/evmc"
+description = "Bindings to EVMC (low-level)"
+categories = ["external-ffi-bindings"]
+edition = "2018"
```
Comment (chfast) [inline bindings/rust/evmc-sys/Cargo.toml]: 2019

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
 /build
 /cmake-build-*
 /.idea
+Cargo.lock
+/target
```
Comment (chfast) [inline .gitignore]: Do you need this top level?

File: bindings/rust/evmc-vm/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "evmc-vm"
+version = "6.2.0-dev"
+authors = ["Alex Beregszaszi <alex@rtfs.hu>"]
+license = "Apache-2.0"
+repository = "https://github.com/ethereum/evmc"
+description = "Bindings to EVMC (VM specific)"
+edition = "2018"
```
Comment (chfast) [inline bindings/rust/evmc-vm/Cargo.toml]: 2019

File: .gitignore (hunk 2 above)
Comment (axic) [reply .gitignore]: If we have the top level `Cargo.toml` then unfortunately it is needed. The reason for having the top level one is the ability to run `cargo build` on the top level.

File: bindings/rust/evmc-sys/Cargo.toml (hunk 1 above)
Comment (axic) [reply bindings/rust/evmc-sys/Cargo.toml]: This is a Rust specific field. 2018 is valid, 2019 isn't.

File: .bumpversion.cfg [hunk 4]
```diff
@@ -24,3 +24,6 @@ serialize = {major}
 search = EVMC_ABI_VERSION = {current_version}
 replace = EVMC_ABI_VERSION = {new_version}
 
+[bumpversion:file:bindings/rust/evmc-sys/Cargo.toml]
```
Comment (chfast) [inline .bumpversion.cfg]: It should contain 
```
search = version = \"{current_version}\"
replace = version = \"{new_version}\"
```
or something...

Comment (axic) [reply .bumpversion.cfg]: `bumpversion` seems to work without these

**Pull request merged**